runtests.py

```diff
@@ -47,6 +47,7 @@
     'self': [executable, '-m', 'mypy', '--config-file', 'mypy_self_check.ini', '-p', 'mypy'],
     # Lint
     'lint': ['flake8', '-j0'],
+    "format": ["black", "."],
```

Comment:
I think for the purposes of rolling this out we may need to do the part where we make it possible to run black in a separate commit from the one where we add CI (which needs to be done alongside the actual blackening). The reason is so that we can rebase old PRs using git by rebasing first on the commit that allows us to run black in the first place.

I don't know anything about github's solution to this, but I'd prefer to not trust it completely and have an option to manually rebase a PR over black if I need to.

Reply:
Currently the PR has two commits, one with just the config, and one with just the formatting.

Don't know if you saw, I had some other ideas about how to best rebase onto this change (in the OP).

Reply:
Rebasing isn't guaranteed to work as well as reparenting.

Reply:
Can you explain what you mean by reparenting? from your instructions it looks as if it effectively squashes the branch into a single diff.

I think that if you do a rebase, always take the changes from that commit, then run black and amend. You will keep the same commit history and everything will be peachy.

Reply:
Hmm I guess I misunderstood that the rebase is only going to go 1 commit forward.